corehq/messaging/scheduling/tasks.py

```diff
@@ -442,7 +442,13 @@ def _handle_schedule_instance(instance, save_function):
             save_function(instance)
             return False
 
-        instance.handle_current_event()
+        try:
+            instance.handle_current_event()
+        except Exception:
+            instance.attempts += 1
+            instance.last_attempt = datetime.utcnow()
+            delay = instance.attempts * 60
+            instance.next_event_due += timedelta(minutes=delay)
```

Comment:
Should this save the instance and re-raise to avoid continuing down the happy path?

Reply:
Yeah, that make sense

Reply:
https://github.com/dimagi/commcare-hq/pull/30758/commits/91fa993fdcfb110b6d863067c2ffec4bada92047